cli/src/packageInstaller.ts

```diff
@@ -0,0 +1,99 @@
+import { readFile } from "node:fs/promises"
+import { join } from "node:path"
+import { spawn } from "node:child_process"
+import { verboseLog } from "./command"
+
+interface PkgJson {
+    dependencies: Record<string, string>
+}
+
+export const isPackageInstalledLocally = async (pkgName: string): Promise<boolean> => {
+    const pkgJsonString = await readFile(join(process.cwd(), "package.json"), { encoding: "utf-8" })
+    const pkgJson = JSON.parse(pkgJsonString) as PkgJson
+
+    return Object.keys(pkgJson.dependencies).includes(pkgName)
+}
+
+const booleanAsk = async (question: string, defValue = true): Promise<boolean> => {
+    let resolve: (val: boolean) => void
+    const defValueString = defValue ? "(Y/n)" : "(y/N)"
+
+    const ask = (q: string) => {
+        process.stdout.write(`${q} ${defValueString}?`)
+    }
+
+    process.stdin.on("data", (data) => {
+        const response = data.toString().trim().toLowerCase()
+
+        if (response === "y") {
+            resolve(true)
+        } else if (response === "n") {
+            process.stdin.end()
+            resolve(false)
+        } else if (response === '') {
+            if (defValue)
+                resolve(true)
+            else
+                resolve(false)
+        }
+        else {
+            ask(`Unknown option: ${response}. Use y/n and try again \n${question}`);
+        }
+    })
+
+    ask(question);
+
+    return new Promise((res) => {
+        resolve = res
+    });
+}
+
+export const spawnAsyncInstaller = (packageName: string) => {
+    let resolve: () => void
+    let reject: () => void
+
+    const installProcess = spawn('npm', ['install', '--save', '--no-workspaces', packageName], {
```

Comment:
I think we have some code to detect users that rely on yarn

Reply:
sure, detection added